yaml/build.go

```diff
@@ -21,3 +22,43 @@ type Build struct {
 	Steps       StepSlice          `yaml:"steps,omitempty"     json:"steps,omitempty" jsonschema:"oneof_required=steps,description=Provide sequential execution instructions.\nReference: https://go-vela.github.io/docs/reference/yaml/steps/"`
 	Templates   TemplateSlice      `yaml:"templates,omitempty" json:"templates,omitempty" jsonschema:"description=Provide the name of templates to expand.\nReference: https://go-vela.github.io/docs/reference/yaml/templates/"`
 }
+
+// ToLibrary converts the Build type to a library Pipeline type.
+func (b *Build) ToLibrary() *library.Pipeline {
```

Comment:
Thoughts on potentially renaming this function to be a bit more clear? Obviously being too verbose is not ideal, but I would expect a `yaml.build.ToLibrary()` function to convert a `yaml.build` to a `library.build`, kinda like how `Template` is done: https://github.com/go-vela/types/blob/5a3c46e550ce7b9969d9acfb843cf1982dd14240/yaml/template.go#L53. 

Maybe `ToPipeline()` or `ToLibraryPipeline()`? It's not ideal either way, but I can see its usage in other repos being confusing.

Reply:
I don't think we can use `ToPipeline()` because those functions already exist today 😞 

* [yaml.Metadata](https://github.com/go-vela/types/blob/master/yaml/metadata.go#L21-L36)
* [yaml.Ruleset](https://github.com/go-vela/types/blob/master/yaml/ruleset.go#L38-L48)
* [yaml.SecretSlice](https://github.com/go-vela/types/blob/master/yaml/secret.go#L45-L64)
* [yaml.ServiceSlice](https://github.com/go-vela/types/blob/master/yaml/service.go#L35-L58)
* [yaml.StageSlice](https://github.com/go-vela/types/blob/master/yaml/stage.go#L31-L50)
* [yaml.StepSlice](https://github.com/go-vela/types/blob/master/yaml/step.go#L42-L69)
* [yaml.UlimitSlice](https://github.com/go-vela/types/blob/master/yaml/ulimit.go#L30-L47)
* [yaml.VolumeSlice](https://github.com/go-vela/types/blob/master/yaml/volume.go#L29-L46)
* [yaml.WorkerSlice](https://github.com/go-vela/types/blob/master/yaml/worker.go#L16-L23)

These functions convert a `github.com/go-vela/types/yaml` type to a `github.com/go-vela/types/pipeline` type.

As for the other suggestion, the only concern I have for `ToLibraryPipeline()` is due to the length of the name.

However, I'll make that change now since I don't have an alternative to suggest 👍 

FWIW:

The naming convention isn't something I was too concerned about here.

I say that because it's not currently possible to convert a `yaml.Build{}` type to a `library.Build{}` type.

And I don't think that is functionality we can or want to add to the product.

Reply:
This is complete 👍